src/commands/analyze/utils.ts

```diff
@@ -162,32 +162,57 @@ export async function addServicesToConfig(configPath: string, services: YAMLServ
+ * into the frontend environment variables like react, vue, angular, nextjs and nuxt.
+ *
  * @param frontendPrefix The prefix to use for the frontend environment variables.
  * @param configPath The path to the config file.
  */
-export async function injectBackendApiUrlsInConfig(configPath: string, frontendPrefix: string) {
+export async function injectBackendUrlsInConfig(configPath: string) {
     const configIOController = new YamlConfigurationIOController(configPath);
 
     // Load config with minimal changes
     const config = await configIOController.read(/* fillDefaults= */ false);
 
-    // Validate backend and frontend existence
-    let functions: string[] = [];
-    const ssrFrameworks: string[] = [];
     const backend = config.backend as YAMLBackend;
-    const frontend = config.frontend as YamlFrontend;
-    if (!frontend) return;
-
-    if (backend?.functions) functions = backend.functions.map((fn) => fn.name);
-    if (config.nextjs) ssrFrameworks.push("nextjs");
-    if (config.nestjs) ssrFrameworks.push("nestjs");
-    if (config.nuxt) ssrFrameworks.push("nuxt");
-    if (config.nitro) ssrFrameworks.push("nitro");
+    const functions: string[] = backend?.functions?.map((fn) => fn.name) || [];
+    const ssrFunctions: string[] = [];
+    if (config.nestjs) ssrFunctions.push("nestjs");
+    if (config.nitro) ssrFunctions.push("nitro");
+
+    // Early return if there is nothing to inject
+    if (functions.length === 0 && ssrFunctions.length === 0) {
+        return;
+    }
 
     // Generate frontend environment variables based on backend functions
-    const frontendEnvironment = createFrontendEnvironment(frontendPrefix, functions, ssrFrameworks);
-    frontend.environment = frontendEnvironment;
+    const frontend = config.frontend as YamlFrontend;
+    if (frontend) {
+        // TODO Support multiple frontend frameworks in the same project
+        const frontendPrefix = getFrontendPrefix(Object.keys(frontend)[0]);
+        const frontendEnvironment = {
+            ...createFrontendEnvironmentRecord(frontendPrefix, functions, ssrFunctions),
+            ...frontend.environment,
+        };
+        frontend.environment = frontendEnvironment;
+        config.frontend = frontend;
+    }
+    // Generate nextjs environment variables based on backend functions
+    if (config.nextjs) {
+        const nextjsEnvironment = {
+            ...createFrontendEnvironmentRecord("NEXT_PUBLIC", functions, ssrFunctions),
+            ...config.nextjs.environment,
+        };
+        config.nextjs.environment = nextjsEnvironment;
+    }
+    // Generate nuxt environment variables based on backend functions
+    if (config.nuxt) {
+        const nuxtEnvironment = {
+            ...createFrontendEnvironmentRecord("NUXT", functions, ssrFunctions),
+            ...config.nuxt.environment,
+        };
+        config.nuxt.environment = nuxtEnvironment;
```

Comment:
I think this approach could be easier to maintain, and the ssrs differs only in env name prefix.

   '''
    const ssrEnvironments = {
        nextjs: "NEXT_PUBLIC",
        nuxt: "NUXT",
    };

    Object.keys(ssrEnvironments).forEach((framework) => {
        if (config[framework]) {
            const prefix = ssrEnvironments[framework];
            config[framework].environment = {
                ...createFrontendEnvironmentRecord(prefix, functions, ssrFunctions),
                ...config[framework].environment,
            };
        }
    });
'''